homeassistant/components/sonos/media_player.py

```diff
@@ -1245,4 +1275,13 @@ def device_state_attributes(self):
         if self._speech_enhance is not None:
             attributes[ATTR_SPEECH_ENHANCE] = self._speech_enhance
 
+        if self._queue is not None:
+            attributes[ATTR_QUEUE] = self._queue
```

Comment:
I don't think this is allowed, it (potentially) adds way too much data to the state attributes.

Reply:
In the pysonos library are the maximum items of the queue hard-coded
```python
def get_queue(self, start=0, max_items=100, full_album_art_uri=False):
    """Get information about the queue.

```
Is this to much?

Reply:
And the queue only holds all titles...

Reply:
This attribute will be a potentially very long string that gets written to the state (and therefore `recorder` database) on any state/attribute change.

Reply:
Where do I have to store the queue then?

Reply:
Is the queue itself useful to expose for scripts/automations? Or is it mostly a UI benefit?

Reply:
It should be a UI benefit

Reply:
I don't think a state attribute is appropriate, especially if there isn't a use case for use in scripts/automations.

It could be later exposed via a websocket call that a future media player card could display in a meaningful way.

Reply:
So I can store the queue in self._queue, but not expose it as an attribute?
Can I expose an attribute with the queue length?

Reply:
Sure, you don't need to expose every instance attribute onto the state object.

A queue length attribute seems reasonable to expose as an attribute if there's any potential use for it.

Reply:
@jjlawren is right that we should not add state if it is only because it would be nice to see in the UI.

I suggest to limit this PR to the `sonos.remove_from_queue` service and the `is_playing_local_queue` attribute (or replace the latter with a queue position or queue length which is `None` when not playing the local queue, if that is more useful for automations). That is a fine first step.

If the rest is moved to a separate PR, that will make it easier to talk about and present matching use cases.